README.md

```diff
@@ -44,46 +42,321 @@ see https://github.com/ioBroker/ioBroker/wiki/Installation
+
+e.g. for Linux:
+```apt-get install redis-server``` .
+
+
+##### Configure Redis
+
+###### Allow Network access
+Ideally the Redis server should be installed on the same host as the js-controller process because as soon as Redis is configured to be used the ioBroker installation will not work without it.
+
+If you plan to install Redis on a different host then the js-controller or use mulithost environment (see below) you must allow connections to redis from any address (default only 127.0.0.1).
 To do that edit file */etc/redis/redis.conf* (```sudo nano /etc/redis/redis.conf```) and replace ```bind 127.0.0.1``` with ```bind 0.0.0.0``` .
-Don't forget to restart redis after that. (```sudo /etc/init.d/redis-server restart```)
+Don't forget to restart redis after that. (```sudo systemctl restart redis-server```)
+
+###### Redis Slaves
+
+If you have multiple hosts you can install Redis to all of them and configure the other hosts to be slaves of the Master host. With this all data are available on all hosts, so even if one host crashs completely you still have the database content from the other hosts.
+
+More details can be found at https://redis.io/topics/replication#configuration and https://redis.io/commands/replicaof
+
+Redis slaves will be not writable by default.
+
+In case of a crash of the master you can reconfigure one of the slaves to be the new master and it will use the current content. Reconfigure all Slaves to sync with the new Master and the logic is restored.
+
+After reconfiguring Redis you also need to update all ioBroker states/objects DB settings to connect to the new Redis Master host.
+
+###### Redis persistence
+
+By default Redis is a pure In-memory Database which means that it has no content on start. In the default configuration the content of the redis DB is stored on disk after a certain amount of changed keys.
 
-To install on windows download latest release here [https://github.com/MSOpenTech/redis/releases](https://github.com/MSOpenTech/redis/releases).
+Depending on how many changes your system is doing it will update the data on disk roughly every 5 minutes by default.
+Please consider this if you run on a SD card.
+
+If you are not working on a SD card and want to have real up-to-date data you can use the second persistance method called AOF.
+
+Please see https://redis.io/topics/persistence for details, differences and configuration information for both persistence options.
+
+##### Using REDIS as States-DB
+**Feature Status: Stable**
+
+There is a possibility to use REDIS as states database. It is reasonable to do that for big installations or for systems with performance problems.
+It is possible to switch anytime between REDIS and In-Memory Javascript DB. 
+
+**Note for js-controller <2.0**: After changing to Redis all states must be updated by the adapters again (values will be lost). Please especially note this for own JavaScript objects! Objects and configuration are not affected.
```

Comment:
Why do we need to note this if objects are not affected?

Reply:
ok ... values of own JavaScript states